app/src/main/java/com/mealsmadeeasy/ui/splash/SplashActivity.kt

```diff
@@ -1,15 +1,122 @@
 package com.mealsmadeeasy.ui.splash
 
 import android.os.Bundle
+import android.content.Intent
+import android.support.v4.app.FragmentActivity
+import android.util.Log
 import com.mealsmadeeasy.ui.BaseActivity
 import com.mealsmadeeasy.ui.home.HomeActivity
+import com.mealsmadeeasy.R
+import android.view.View
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.android.gms.tasks.Task
+
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.GoogleAuthProvider
+import android.support.design.widget.Snackbar
+import android.widget.Button
+import com.google.android.gms.common.SignInButton
+import kotlinx.android.synthetic.main.activity_splash.*
+
+
+class SplashActivity : BaseActivity(), View.OnClickListener {
```

Comment:
This code probably needs to exist in a different Activity. The point of SplashActivity is to be a really simple screen that just routes to whichever screen the user should see first. So for our app, it should check if the user is already logged in. If so, close this activity and open the week at a glance screen. If not, it should close this activity and start a `LoginActivity`, which is where this code should live.

If you don't do this, then the user will briefly see the login screen every time they start the app.